Tools/Scripts/pgo-profile-merge

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+
+import argparse
+import subprocess
+import os
+
+def pad(string, maxLength):
+    if len(string) > maxLength:
+        return string[:maxLength - 1] + u"…"
+    return string.ljust(maxLength)
+
+def shell(command):
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    return process.communicate()[0].strip()
+
+def shortened(name):
+    return name if len(name) < 200 else name[0:99] + u"…" + name[-98:]
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Merge a pile of *.profraw files into the *.profdata files we can build with.")
+    parser.add_argument("directory", help="Path to the directory containing the *.profraw files")
+    args = parser.parse_args()
+    if os.path.isdir(args.directory):
+        profilePath = args.directory
+    else:
+        print("No folder at %s" % args.directory)
+        exit()
+
+    for lib in ["JavaScriptCore", "WebCore", "WebKit"]:
+        print(shell(f"xcrun -sdk macosx.internal llvm-profdata merge {args.directory}/{lib}*.profraw -output={args.directory}/{lib}.profdata"))
```

Comment:
I'd recommend using `subprocess.check_output`, which will do what your `shell` function does plus error handling. Also, this will fail if `args.directory` contains spaces, and can do arbitrary code execution if it contains shell control characters.

It's probably find to surround `{args.directory}` with quotes to handle the spaces issue. If you wanna do something more resilient, look at the `glob` module or the higher-level path API in Pathlib. Then you could call subprocess without `shell=True`.